app/controllers/admin/purchases_controller.rb

```diff
@@ -52,8 +52,10 @@ def resend_receipt
       @purchase.resend_receipt
       render json: { success: true }
     else
-      render json: { success: false }
+      render json: { success: false }, status: :not_found
```

Comment:
It still makes sense to rendered a failed response (here, not found) so the red flash message shows up instead of nothing

Reply:
I guess so, though I discovered there is a `e404_json` method for JSON 404 responses, so eventually I went with it for consistency. I'm still removing the `success: false` from the 422 response though: https://github.com/antiwork/gumroad/pull/1988/commits/0b41712ab59d471fb8d5b21f91c5694714a41063